### PR TITLE
Linux: Make zfs_prune() fair on NUMA systems

### PR DESCRIPTION
**Pull request by amotin**
Previous code evicted nr_to_scan items from each NUMA node.  This not only multiplied the eviction by the number of nodes, but could exhaust the smaller ones, evicting inodes used by active workload and requiring their immediate recreation.  This patch spreads the requested eviction between all NUMA nodes proportionally to their evictable counts, which should be closer to expected LRU logic. See kernel's super_cache_scan() as a similar logic example.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: module/os/linux/zfs/zfs_vfsops.c [hunk 1]
```diff
@@ -1264,14 +1264,22 @@ zfs_prune(struct super_block *sb, unsigned long nr_to_scan, int *objects)
 	defined(SHRINK_CONTROL_HAS_NID) && \
 	defined(SHRINKER_NUMA_AWARE)
 	if (shrinker->flags & SHRINKER_NUMA_AWARE) {
+		long tc = 1;
+		for_each_online_node(sc.nid) {
+			long c = shrinker->count_objects(shrinker, &sc);
+			if (c  == 0 || c == SHRINK_EMPTY)
```
Comment (ixhamza) [inline module/os/linux/zfs/zfs_vfsops.c:1270]: Do we need to check for `SHRINK_EMPTY` since I do not see it being returned from `arc_shrinker_count()`.

Comment (amotin) [reply module/os/linux/zfs/zfs_vfsops.c:1270]: That is what kernel does, I've copied form there.

Comment (amotin) [reply module/os/linux/zfs/zfs_vfsops.c:1270]: PS: And we are not calling arc_shrinker_count() here, but a kernel's superblock one.

Comment (ixhamza) [reply module/os/linux/zfs/zfs_vfsops.c:1270]: Thanks for the clarification.

**Review by behlendorf — APPROVED** [review]

**Pull request merged**